### DISCUSSION TITLE
Can I prevent guest editors from viewing other users entries?

### DISCUSSION THREAD

**Discussion by therealpaperclip** [statamic/cms#5166] [Q&A] [answered]
Hi all,

I'm evaluating Statamic for a use case where it would be useful that other users do not see entries of other users. "Guest editors" should only be able to create new entries and only view their own entries. Is something like that possible / controllable?

Regards

**Comment by ryanmitchell** [comment, marked as answer]
I think there is a permission for exactly that - you would just set a role with that permission and assign guest editors to that role.

**Reply by therealpaperclip** [reply]
Hi @ryanmitchell , thanks for getting back to me. I already made myself familiar with the permission system. It appears, that all entries are viewable to all users while you can restrict access to read-only so that only authors of their posts are able to edit them while other Guest authors cannot. But it seems not possible to also hide non-owned entries.

**Reply by ryanmitchell** [reply]
So in your App\Providers\AppServiceProvider.php add something like this to your boot()

`$this->app->bind(\Statamic\Http\Controllers\CP\Collections\EntriesController::class, \App\Http\Controllers\FilterEntriesController::class);`

then create App\Http\Controllers\FilterEntriesController.php something like the following

```
<?php

namespace App\Http\Controllers;
		
use Statamic\Facades\User;
use Statamic\Http\Controllers\CP\Collections\EntriesController;

class FilterEntriesController extends EntriesController
{
	protected function indexQuery($collection)
	{
		$query = parent::indexQuery($collection);
		
		// you might want to check user roles (User::current()->hasRole()) before doing this as it will apply to all users
		$query->where('author', User::current()->id());
				
		return $query;
	}
}
```
